packages/engine/Source/Scene/CameraEventAggregator.js

```diff
@@ -219,20 +218,73 @@ function listenMouseButtonDownUp(aggregator, modifier, type) {
 
   aggregator._eventHandler.setInputAction(
     function () {
-      aggregator._buttonsDown = Math.max(aggregator._buttonsDown - 1, 0);
-      isDown[key] = false;
-      releaseTime[key] = new Date();
+      cancelMouseDownAction(getKey(type, undefined), aggregator);
+      for (const modifierName in KeyboardEventModifier) {
+        if (KeyboardEventModifier.hasOwnProperty(modifierName)) {
+          const modifier = KeyboardEventModifier[modifierName];
+          if (defined(modifier)) {
+            const cancelKey = getKey(type, modifier);
+            cancelMouseDownAction(cancelKey, aggregator);
+          }
+        }
+      }
     },
     up,
     modifier
   );
 }
 
+function cancelMouseDownAction(cancelKey, aggregator) {
+  const releaseTime = aggregator._releaseTime;
+  const isDown = aggregator._isDown;
+  if (isDown[cancelKey]) {
+    aggregator._buttonsDown = Math.max(aggregator._buttonsDown - 1, 0);
+  }
+  isDown[cancelKey] = false;
+  releaseTime[cancelKey] = new Date();
+}
+
 function cloneMouseMovement(mouseMovement, result) {
   Cartesian2.clone(mouseMovement.startPosition, result.startPosition);
   Cartesian2.clone(mouseMovement.endPosition, result.endPosition);
 }
 
+function refreshMouseDownStatus(type, modifier, aggregator) {
+  // first: Judge if the mouse is pressed
+  const isDown = aggregator._isDown;
+  let downFlag = false;
```

Comment:
Can we rename this to something slightly more descriptive? Like `anyButtonIsDown`